packages/venia-ui/lib/components/CategoryTree/categoryTree.js

```diff
@@ -17,33 +17,20 @@ const Tree = props => {
         updateCategories
     } = props;
 
-    const classes = mergeClasses(defaultClasses, props.classes);
-    const [queryResult, queryApi] = useQuery(MENU_QUERY);
-    const { data } = queryResult;
-    const { runQuery } = queryApi;
-
-    // fetch categories
-    useEffect(() => {
-        if (categoryId != null) {
-            runQuery({ variables: { id: categoryId } });
-        }
-    }, [categoryId, runQuery]);
-
-    // update redux with fetched categories
-    useEffect(() => {
-        if (data && data.category) {
-            updateCategories(data.category);
-        }
-    }, [data, updateCategories]);
+    const talonProps = useCategoryTree({
+        categories,
+        categoryId,
+        query: MENU_QUERY,
+        updateCategories
+    });
 
-    const rootCategory = categories[categoryId];
-    const { children } = rootCategory || {};
+    const { childCategories } = talonProps;
+    const classes = mergeClasses(defaultClasses, props.classes);
 
     // for each child category, render a direct link if it has no children
     // otherwise render a branch
-    const branches = Array.from((rootCategory && children) || [], id => {
-        const category = categories[id];
-        const isLeaf = category.children_count === '0';
+    const branches = Array.from(childCategories, childCategory => {
```

Comment:
I like that you made this more clear -- we iterate over children categories rather than the `rootCategory && children`. That said, can we just use `forEach` instead of `Array.from`?

Reply:
Hm? To be clear, this code is mapping one iterable (a `Map` of objects) to another (an array of React elements). `Array.from` is the right choice here because it's a mapping function that works on any iterable, which is what we need when working with a `Map`.

Reply:
Oh, yes of course -- you need to return the iterable. ignore me :D